version/version.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2018, Google, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package version
+
+import (
+	"fmt"
+)
+
+// The git commit that was compiled. This is filled in by the compiler.
+var GitCommit string
+
+// Main version number being run right now.
+var Version = "0.4.0"
+
+// A pre-release marker for the version. If this is an empty string,
+// then the release is a final release. Otherwise this is a pre-release
+// version e.g. "dev", "alpha", etc.
+var Prerelease = "dev"
```

Comment:
Why not give this a name like ReleasePhase? You can either special case "final" to encode as "" or special case "" to mean final. If you go the encoding route, you can have a function that returns the encoding decorator for the release phase (which would either be "" or "-<phase>" depending on phase being "final" or not)

Reply:
`ReleasePhase` makes sense.